packages/integration-sdk-cli/src/generator/template/src/steps/constants.ts.hbs

```diff
@@ -4,25 +4,23 @@ import {
   StepRelationshipMetadata,
 } from '@jupiterone/integration-sdk-core';
 
-export const Steps: Record<{{#unless steps}}string{{/unless}}{{#each steps}}'{{constantCase name}}'{{#unless @last}}|{{/unless}}{{/each}}, string> = {
+export const Steps = {
   {{#each steps}}
   {{constantCase name}}: '{{kebabCase name}}',
   {{/each}}
-};
+} satisfies Record<string, string>;
```

Comment:
If we switch to `Record<string, string>` from the above will we lose any type information. I assumed the type hinting and suggestions came from the `Record<string,  'STEP_1' | 'STEP_2'>`

Reply:
Actually `satisfies` fix this problem, it will validate if the returned type from the object matches the Record type, but return the original type. We have been using this in new integrations.
![image](https://github.com/JupiterOne/sdk/assets/4943174/ac52fc25-e708-4104-b2e3-94d6be75a4ca)